interfaces/acados_matlab_octave/AcadosMultiphaseOcp.m

```diff
@@ -0,0 +1,402 @@
+        constraints
+        solver_options
+        mocp_opts
+        dummy_ocp_list
+
+        model
+        parameter_values % initial value of the parameter
+        problem_class
+        simulink_opts
+        name
+
+        cython_include_dirs
+        code_export_directory
+        json_file
+        shared_lib_ext
+        acados_include_path
+        acados_lib_path
+
+        % detected fields
+        start_idx
+        end_idx
+        cost_start_idx
+
+    end
+    methods
+        function obj = AcadosMultiphaseOcp(N_list)
+            if any(N_list < 1)
+                error('N_list must be a list of positive integers');
+            end
+            n_phases = length(N_list);
+            obj.n_phases = n_phases;
+            obj.N_list = N_list;
+            obj.N_horizon = sum(N_list);
+
+            obj.phases_dims = cell(n_phases, 1);
+            obj.cost = cell(n_phases, 1);
+            obj.constraints = cell(n_phases, 1);
+            obj.model = cell(n_phases, 1);
+
+            for i=1:n_phases
+                obj.phases_dims{i} = AcadosOcpDims();
+                obj.cost{i} = AcadosOcpCost();
+                obj.constraints{i} = AcadosOcpConstraints();
+                obj.model{i} = AcadosModel();
+            end
+
+            self.dummy_ocp_list = cell(n_phases, 1);
+
+            obj.solver_options = AcadosOcpOptions();
+            obj.solver_options.N_horizon = obj.N_horizon; % NOTE: to not change options when making ocp consistent
+
+            obj.mocp_opts = AcadosMultiphaseOptions();
+
+            obj.parameter_values = cell(n_phases, 1);
+            obj.problem_class = 'MOCP';
+            obj.simulink_opts = [];
+            obj.cython_include_dirs = [];
+            obj.json_file = 'mocp.json';
+            obj.shared_lib_ext = '.so';
+            obj.name = 'ocp';
+            if ismac()
+                obj.shared_lib_ext = '.dylib';
+            end
+            obj.code_export_directory = 'c_generated_code';
+
+            % set include and lib path
+            acados_folder = getenv('ACADOS_INSTALL_DIR');
+            obj.acados_include_path = [acados_folder, '/include'];
+            obj.acados_lib_path = [acados_folder, '/lib'];
+        end
+
+
+        function set_phase(self, ocp, phase_idx)
+            % Note: phase_idx is 1-indexed in contrast to Python!
+            if phase_idx > self.n_phases
+                error('phase_idx must be less than or equal to the number of phases');
+            end
+
+            % Check solver options
+            non_default_opts = find_non_default_fields_of_obj(ocp.solver_options);
+
+            if ~isempty(non_default_opts)
+                fprintf('WARNING: set_phase: Phase %d contains non-default solver options: %s, which will be ignored.\n', ...
+                        phase_idx, strjoin(non_default_opts, ', '));
+                fprintf('Solver options need to be set via AcadosMultiphaseOcp.solver_options or mocp_opts instead.\n');
```

Comment:
maybe better: 'The solver option %s needs to be set via ...' to make it more explicit that not all options are in the MOCP Options.

Reply:
I improved the message.
However this is just a warning, if the phase OCP is set up in a way that also options are set, I want to only print one warning for all options, as done now.
It is the same in python.